src/components/WalletConnect/NanoContract/NewNanoContractTransactionRequest.js

```diff
@@ -0,0 +1,313 @@
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {
+  useEffect,
+  useMemo,
+  useState
+} from 'react';
+import {
+  StyleSheet,
+  View,
+  ScrollView,
+  TouchableWithoutFeedback,
+  Image
+} from 'react-native';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigation } from '@react-navigation/native';
+import { t } from 'ttag';
+import {
+  nanoContractBlueprintInfoRequest,
+  setNewNanoContractStatusReady,
+  tokensFetchMetadataRequest,
+  walletConnectAccept,
+  walletWalletReject
+} from '../../../actions';
+import { COLORS } from '../../../styles/themes';
+import NewHathorButton from '../../NewHathorButton';
+import { SelectAddressModal } from '../../NanoContract/SelectAddressModal';
+import { FeedbackContent } from '../../FeedbackContent';
+import { DEFAULT_TOKEN, WALLETCONNECT_NEW_NANOCONTRACT_TX_STATUS } from '../../../constants';
+import Spinner from '../../Spinner';
+import FeedbackModal from '../../FeedbackModal';
+import errorIcon from '../../../assets/images/icErrorBig.png';
+import checkIcon from '../../../assets/images/icCheckBig.png';
+import { DappContainer } from './DappContainer';
+import { NanoContractExecInfo } from './NanoContractExecInfo';
+import { NanoContractActions } from './NanoContractActions';
+import { NanoContractMethodArgs } from './NanoContractMethodArgs';
+import { DeclineModal } from './DeclineModal';
+
+/**
+ * @param {Object} props
+ * @param {Object} props.ncTxRequest
+ * @param {Object} props.ncTxRequest.nc
+ * @param {Object} props.ncTxRequest.dapp
+ * @param {string} props.ncTxRequest.dapp.icon
+ * @param {string} props.ncTxRequest.dapp.proposer
+ * @param {string} props.ncTxRequest.dapp.url
+ * @param {string} props.ncTxRequest.dapp.description
+ */
+export const NewNanoContractTransactionRequest = ({ ncTxRequest }) => {
+  const { nc, dapp } = ncTxRequest;
+  const dispatch = useDispatch();
+  const navigation = useNavigation();
+  const newTxStatus = useSelector((state) => state.walletConnect.newNanoContractTransaction.status);
+  const firstAddress = useSelector((state) => state.firstAddress);
+  // Nullable if the nano contract method is 'initialize'
+  const registeredNc = useSelector((state) => state.nanoContract.registered[nc.ncId]);
+  const registeredTokensMetadata = useSelector((state) => state.tokenMetadata);
+  // Use it to add loading feedback
+  const metadataLoaded = useSelector((state) => state.metadataLoaded);
+  const blueprintInfo = useSelector((state) => state.nanoContract.blueprint[nc.blueprintId]);
+
+  const [showSelectAddressModal, setShowSelectAddressModal] = useState(false);
+  const [showDeclineModal, setShowDeclineModal] = useState(false);
+  // If nano-contract's method is 'initialize' then the expression
+  // should be resolved to firstAddress value by default.
+  // However, the wallet can have trouble to
```

Comment:
Trouble to what?

Reply:
Never mind. The user will see a feedback message instructing it to select an address for the transaction in case the first address request fails.